dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -416,6 +416,50 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
     end
   end
 
+  test 'login: oauth takeover takes over account when account has no activity' do
+    User::OAUTH_PROVIDERS_UNTRUSTED_EMAIL.each do |provider|
+      oauth_student = create :student, provider: provider, uid: '12345'
+      student = create :student
+
+      @request.session[ACCT_TAKEOVER_EXPIRATION] = 5.minutes.from_now
+      @request.session[ACCT_TAKEOVER_PROVIDER] = provider
+      @request.session[ACCT_TAKEOVER_UID] = oauth_student.uid
+      @request.session[ACCT_TAKEOVER_OAUTH_TOKEN] = '54321'
+
+      check_and_apply_oauth_takeover(student)
+
+      oauth_student.reload
+      refute_nil oauth_student.deleted_at
+      assert_equal provider, student.provider
+      assert_equal oauth_student.uid, student.uid
+      assert_equal '54321', student.oauth_token
+      assert_nil @request.session['clever_link_flag']
+    end
+  end
+
+  test 'login: oauth takeover does nothing if account has activity' do
+    User::OAUTH_PROVIDERS_UNTRUSTED_EMAIL.each do |provider|
+      oauth_student = create :student, provider: provider, uid: '12345'
+      student = create :student
+      level = create(:level)
+      create :user_level, user: oauth_student, level: level, attempts: 1, best_result: 1
+
+      assert oauth_student.has_activity?
+
+      Honeybadger.expects(:notify).at_least_once
+
+      @request.session[ACCT_TAKEOVER_EXPIRATION] = 5.minutes.from_now
+      @request.session[ACCT_TAKEOVER_PROVIDER] = provider
+      @request.session[ACCT_TAKEOVER_UID] = oauth_student.uid
+      @request.session[ACCT_TAKEOVER_OAUTH_TOKEN] = '54321'
+
+      check_and_apply_oauth_takeover(student)
+
+      assert_nil oauth_student.deleted_at
```

Comment:
I think we ought to call `oauth_student.reload` before checking deleted_at here.

Reply:
Ah, good call, I used to have that, but I deleted it (it doesn't make the test fail, but it does make the check irrelevant)